openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/ConceptResource.java

```diff
@@ -96,10 +100,17 @@ protected Concept retrieveImpl(String uuid, RequestContext context, long snapsho
             for (FormField formField : chart.getFormFields()) {
                 Field field = formField.getField();
                 Concept fieldConcept = field.getConcept();
-                if (fieldConcept == null) continue;
-                ret.add(fieldConcept);
-                for (ConceptAnswer answer : fieldConcept.getAnswers(false)) {
-                    ret.add(answer.getAnswerConcept());
+                if (fieldConcept != null) {
+                    ret.add(fieldConcept);
+                    for (ConceptAnswer answer : fieldConcept.getAnswers(false)) {
+                        ret.add(answer.getAnswerConcept());
+                    }
+                }
+                // Fetch and add additional concepts that might have been stored in the field
+                // description.
+                for (Concept additionalConcept :
+                        getConceptsFromFieldDescription(field.getDescription())) {
+                    ret.add(additionalConcept);
```

Comment:
For most fields, which only have a single concept, this additional code is an expensive no-op, but because we've got no way of knowing how many concepts each field references until we parse the JSON we have to live with the cost anyway. 

We can't remove the code that adds the answer concepts though, because they're not necessarily included in the JSON in the `description` field.

Does that answer your question? I'm not sure I understand.